### PR TITLE
[PENG-1676] Add re-submit endpoint to API's job_submissions router

### PR DESCRIPTION
**Pull request by fschuch**
#### What
* Add a 're-submit' endpoint to the job_submissions routerre-submission
* Add unit tests
* Update documentation

#### Why
The jobbergate-api needs an endpoint that allows a client to re-submit a job_submission with exactly the same parameters as a previous submission.

`Tasks`:
* https://app.clickup.com/t/18022949/PENG-1676
* https://jira.scania.com/browse/ASP-4597

---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

File: jobbergate-api/jobbergate_api/apps/job_submissions/routers.py [hunk 1]
```diff
@@ -87,6 +88,31 @@ async def job_submission_create(
...
+    id: int = Path(...),
+    secure_services: SecureService = Depends(
+        secure_services(Permissions.ADMIN, Permissions.JOB_SUBMISSIONS_CREATE, ensure_email=True)
+    ),
+):
+    """Clone a job_submission given its id."""
+    logger.info(f"Cloning job submission {id=}")
+
+    original_instance = await secure_services.crud.job_submission.get(id)
+    cloned_instance = await secure_services.crud.job_submission.clone_instance(
```
Comment (fschuch) [inline jobbergate-api/jobbergate_api/apps/job_submissions/routers.py:107]: Set status to crated to ensure they are processed by the agent

File: jobbergate-api/jobbergate_api/apps/job_submissions/routers.py [hunk 2]
```diff
@@ -87,6 +88,31 @@ async def job_submission_create(
     return new_job_submission
 
 
+@router.post(
+    "/clone/{id}",
+    status_code=status.HTTP_201_CREATED,
+    response_model=JobSubmissionDetailedView,
+    description="Endpoint for cloning a job script to a new entry owned by the user",
```
Comment (dusktreader) [inline jobbergate-api/jobbergate_api/apps/job_submissions/routers.py:95]: The description should probably let the caller know that this is used for resubmission and that the new, cloned submission will be submitted to the cluster.

**Review by dusktreader — COMMENTED** [review]
Looks good so far!

File: jobbergate-api/jobbergate_api/apps/services.py [hunk 3]
```diff
@@ -231,6 +231,11 @@ async def clone_instance(self, original_instance: CrudModel, **incoming_data) ->
         data.update(incoming_data)
         data["cloned_from_id"] = original_instance.id
 
+        if "created_at" in data:
+            del data["created_at"]
+        if "updated_at" in data:
+            del data["updated_at"]
```
Comment (dusktreader) [inline jobbergate-api/jobbergate_api/apps/services.py:237]: You could use `pop()` here to do these operations in one line:

```suggestion
        data.pop("created_at", None)
        data.pop("updated_at", None)
```

File: jobbergate-api/tests/apps/job_submissions/test_routers.py [hunk 4]
```diff
@@ -350,6 +350,88 @@ async def test_create_job_submission_without_client_id(
...
+
+    assert response.status_code == status.HTTP_201_CREATED, f"Clone failed: {response.text}"
+    response_data = response.json()
+
+    assert response_data["cloned_from_id"] == original_instance.id
+    assert response_data["id"] != original_instance.id
+    assert response_data["owner_email"] == new_owner_email
+    assert response_data["slurm_job_id"] is None
+    assert response_data["status"] == JobSubmissionStatus.CREATED
+
```
Comment (dusktreader) [inline jobbergate-api/tests/apps/job_submissions/test_routers.py:400]: I think you need to include checks here for the timestamps being updated.

File: jobbergate-api/jobbergate_api/apps/services.py (hunk 3 above)
Comment (fschuch) [reply jobbergate-api/jobbergate_api/apps/services.py]: Solved by [7883b10](https://github.com/omnivector-solutions/jobbergate/pull/643/commits/7883b10a9462d051e4dd414c781c958d4e9350a5)

File: jobbergate-api/tests/apps/job_submissions/test_routers.py (hunk 4 above)
Comment (fschuch) [reply jobbergate-api/tests/apps/job_submissions/test_routers.py:403]: Solved by [7883b10](https://github.com/omnivector-solutions/jobbergate/pull/643/commits/7883b10a9462d051e4dd414c781c958d4e9350a5)

**Review by dusktreader — APPROVED** [review]

**Pull request merged**